frame/contracts/src/exec.rs

```diff
@@ -510,9 +502,35 @@ where
 		Ok((dest, output))
 	}
 
-	fn new_call_context<'b>(&'b mut self, caller: T::AccountId, value: BalanceOf<T>)
-		-> CallContext<'b, 'a, T, V, L>
-	{
+	pub fn terminate(
+		&mut self,
+		beneficiary: &T::AccountId,
+		gas_meter: &mut GasMeter<T>,
+	) -> Result<(), DispatchError> {
+		let self_id = self.self_account.clone();
+		let value = self.overlay.get_balance(&self_id);
+		if let Some(parent) = self.parent {
+			if parent.is_live(&self_id) {
+				return Err(DispatchError::Other("Cannot terminate a live contract"));
```

Comment:
I think the users might be confused by this. 

How about "Cannot terminate a contract that is present on the call stack"?

Reply:
Your are absolutely right. Nobody knows what "live" means. The function name `is_live` is equally confusing (but this is not the point here).